server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -762,58 +792,228 @@ void cancelPollingClusterFormationInfo() {
+                    }
+                }
+            } else {
+                Scheduler.Cancellable cancellable = cancellableReference.get();
+                if (cancellable != null) {
+                    cancellable.cancel();
+                }
+            }
+        };
+    }
+
+    /**
+     * This method returns quickly, but in the background schedules to query the remote masterEligibleNode's cluster diagnostics in 10
+     * seconds unless cancel() is called on the Cancellable that this method returns.
+     * @param masterEligibleNode The masterEligibleNode to poll for cluster diagnostics. This masterEligibleNode can be null in the case
+     *                           when there are not yet any master-eligible nodes known to this masterEligibleNode's PeerFinder.
+     * @param responseConsumer The consumer of the cluster diagnostics for the masterEligibleNode, or the exception encountered while
+     *                         contacting it
+     * @return A Cancellable for the task that is scheduled to fetch cluster diagnostics
+     */
+    private Scheduler.Cancellable fetchCoordinationDiagnostics(
+        @Nullable DiscoveryNode masterEligibleNode,
+        Consumer<RemoteMasterHealthResult> responseConsumer
+    ) {
+        return fetchRemoteObject(
+            masterEligibleNode,
+            responseConsumer,
+            CoordinationDiagnosticsAction.INSTANCE,
+            new CoordinationDiagnosticsAction.Request(true),
+            e -> new RemoteMasterHealthResult(masterEligibleNode, null, e),
+            response -> new RemoteMasterHealthResult(masterEligibleNode, response.getCoordinationDiagnosticsResult(), null),
+            Version.V_8_4_0
+        );
+    }
+
+    /**
+     * This method connects to masterEligibleNode and sends it a transport request for a remote response of type R. The remote response or
+     * exception are transformed into a common type T with responseToResultFunction or exceptionToResultFunction, and then consumed by
+     * responseConsumer.
+     *
+     * @param masterEligibleNode        The master eligible node to be queried, or null if we do not yet know of a master eligible node
+     * @param responseConsumer          The consumer of the transformed response
+     * @param transportActionType       The ActionType for the transport action
+     * @param transportActionRequest    The ActionRequest to be sent
+     * @param exceptionToResultFunction A function that converts a remote exception to the response type expected by the responseConsumer
+     * @param responseToResultFunction  A function that converts a remote response to the response type expected by the responseConsumer
+     * @param minSupportedVersion       The minimum version that the transport action works with
+     * @return A Cancellable for the task that is scheduled to fetch the remote information
+     */
+    private <R extends ActionResponse, T> Scheduler.Cancellable fetchRemoteObject(
```

Comment:
Is `Object` a bit too generic here? (and possibly overlapping the `java.lang.Object` concept?) 

Should we call it something along the lines of `sendNodeRequest` (as the parameters do the rest in documenting what it's actually fetching) or `fetchRemoteDiagnosticInfo` ?